src/common/reducers/snaps.js

```diff
@@ -0,0 +1,41 @@
+import * as ActionTypes from '../actions/snaps';
+import { parseGitHubRepoUrl } from '../helpers/github-url';
+
+export function snaps(state = {
+  isFetching: false,
+  success: false,
+  error: null,
+  snaps: null,
+}, action) {
+  switch(action.type) {
+    case ActionTypes.FETCH_SNAPS:
+      return {
+        ...state,
+        isFetching: true
+      };
+    case ActionTypes.FETCH_SNAPS_SUCCESS:
+      return {
+        ...state,
+        isFetching: false,
+        success: true,
+        snaps: action.payload.map((snap) => {
+          return {
+            // parse repository info to keep consistent data format
+            repo: parseGitHubRepoUrl(snap.git_repository_url),
```

Comment:
Every snap on the list contains two pieces of information now:
```js
{
  repo: { fullName, owner, name, url } // full repo object as used elsewhere in the app
  snap: { git_repository_url, ... } // full snap data as returned from API
}
```

Maybe we don't need whole snap object here, but we will likely need some API URLs from there (to fetch build status, etc).